c10/cuda/CUDACachingAllocator.h

```diff
@@ -98,7 +98,7 @@ struct DeviceStats {
   Stat oversize_segments;
 
   // SIZE: maximum block size that is allowed to be split.
-  int64_t max_split_size = 0;
+  size_t max_split_size = 0;
```

Comment:
To avoid breaking this again, 

should a unit test to be added?
A test case where -1 could be returned.

Reply:
@cyyever said in the PR description that according to the docs and usage in the code, negative values don't make sense.

Reply:
@kit1980 You said that the test failed in https://github.com/pytorch/pytorch/pull/111137 had passed, so is there a new test failing on this change? Any hints?

Reply:
Another, but related thing came up. @malfet can help on re-land, as I said.

Reply:
I think the same test actually failed, but was not reported correctly. But yes, testing that it should stay as `-1` is easy, though I'm not sure that negative values are invalid in this case.